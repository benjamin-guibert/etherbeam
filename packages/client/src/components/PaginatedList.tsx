import React, { ReactElement, ReactNode } from 'react'
import { Pagination as PaginationType } from 'libraries/api'
import Pagination from './Pagination'

interface PaginatedListProps {
  pagination: PaginationType
  setPage: (next: number | ((previous: number) => number)) => void
  children?: ReactNode
}

const PaginatedList = ({ pagination, setPage, children }: PaginatedListProps): ReactElement => {
  const { totalPages, currentPage } = pagination || {}

  const PaginationComponent = (): ReactElement => (
    <>{totalPages > 1 && <Pagination total={totalPages} current={currentPage} setCurrent={setPage} />}</>
  )

  return (
    <>
      <PaginationComponent />
      {children}
      <PaginationComponent />
    </>
  )
}

export default PaginatedList
