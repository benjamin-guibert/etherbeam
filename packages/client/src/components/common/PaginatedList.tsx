import React, { ReactElement, ReactNode } from 'react'
import { Pagination as PaginationType } from 'libraries/api'
import Pagination from './Pagination'

interface PaginatedListProps {
  pagination: PaginationType
  setPage: (next: number | ((previous: number) => number)) => void
  children: ReactNode
}

const PaginatedList = ({ pagination, setPage, children }: PaginatedListProps): ReactElement => {
  const { totalPages, currentPage } = pagination || {}

  const PaginationComponent = () => (
    <>
      {totalPages > 1 && (
        <Pagination
          className="justify-content-center my-2"
          size="sm"
          total={totalPages}
          current={currentPage}
          setCurrent={setPage}
        />
      )}
    </>
  )

  return (
    <>
      <PaginationComponent />
      {children}
      <PaginationComponent />
    </>
  )
}

export default PaginatedList
