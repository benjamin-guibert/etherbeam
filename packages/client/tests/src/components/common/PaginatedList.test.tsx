import * as React from 'react'
import { render } from 'enzyme'
import PaginatedList from 'components/common/PaginatedList'

describe('<PaginatedList />', () => {
  let component: unknown

  beforeAll(() => {
    component = render(
      <PaginatedList setPage={null} pagination={{ currentPage: 1, itemsPerPage: 10, totalItems: 100, totalPages: 10 }}>
        Content
      </PaginatedList>
    )
  })

  it('should match snapshot', () => expect(component).toMatchSnapshot())
})
