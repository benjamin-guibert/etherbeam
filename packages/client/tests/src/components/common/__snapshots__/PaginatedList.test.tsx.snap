// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginatedList /> should match snapshot 1`] = `
Array [
  <ul
    class="justify-content-center my-2 pagination pagination-sm"
  >
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          class="sr-only"
        >
          First
        </span>
      </span>
    </li>
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          class="sr-only"
        >
          Previous
        </span>
      </span>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        1
        <span
          class="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          …
        </span>
        <span
          class="sr-only"
        >
          More
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        10
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          class="sr-only"
        >
          Next
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          class="sr-only"
        >
          Last
        </span>
      </a>
    </li>
  </ul>,
  "Content",
  <ul
    class="justify-content-center my-2 pagination pagination-sm"
  >
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          class="sr-only"
        >
          First
        </span>
      </span>
    </li>
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          class="sr-only"
        >
          Previous
        </span>
      </span>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        1
        <span
          class="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
        disabled=""
      >
        <span
          aria-hidden="true"
        >
          …
        </span>
        <span
          class="sr-only"
        >
          More
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        10
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          class="sr-only"
        >
          Next
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          class="sr-only"
        >
          Last
        </span>
      </a>
    </li>
  </ul>,
]
`;
